### PR TITLE
Escape like parameters in database user backend

### PR DESCRIPTION
**Pull request by LukasReschke**
This leads to unexpected behaviour otherwise and pretty sure also some very confusing bugs.

@icewind1991 Objections?
@daita Should also fix your issue :)

### CODE REVIEW

File: lib/private/User/Database.php [hunk 1]
```diff
@@ -76,6 +79,7 @@ class Database extends Backend implements IUserBackend {
 	public function __construct($eventDispatcher = null) {
 		$this->cache = new CappedMemoryCache();
 		$this->eventDispatcher = $eventDispatcher ? $eventDispatcher : \OC::$server->getEventDispatcher();
+		$this->connection = \OC::$server->getDatabaseConnection();
```
Comment (LukasReschke) [inline lib/private/User/Database.php]: Let's see if we can move that to DI. Want @icewind1991's feedback however first if this is the right approach.

Comment (LukasReschke) [inline lib/private/User/Database.php]: I bet however that the dependency is too early, thus also the failing tests. But if I get told that this is the right function to use I can adjust our code to work around that :)

**Comment by daita** [comment]
to anyone wondering: 

> 21:06 <@maxence> is it a known feature that when you type the name of a user/group when sharing, you can 'search' by replacing any char by an underscore ?
> 21:06 <@maxence> you can find the user 'test' by writing t__t
>

**Review by nickvergessen — DISMISSED** [review]

File: lib/private/User/Database.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/User/Database.php]: Yes function usage looks good

**Review by icewind1991 — APPROVED** [review]

**Comment by LukasReschke** [comment]
DI doesn't really seem to work here but at least the tests should run now 😄

**Review by rullzer — APPROVED** [review]

**Pull request merged**